packages/@uppy/companion/src/server/Uploader.js

```diff
@@ -47,15 +46,18 @@ class Uploader {
 
     this.options = options
     this.token = uuid.v4()
-    this.options.path = `${this.options.pathPrefix}/${Uploader.FILE_NAME_PREFIX}-${this.token}`
+    this.path = `${this.options.pathPrefix}/${Uploader.FILE_NAME_PREFIX}-${this.token}`
+    this.metadata = Object.assign({}, this.options.metadata || {})
+    this.metadata.filename = this.metadata.name || path.basename(this.path)
+    this.metadata.filetype = this.metadata.type
     this.streamsEnded = false
```

Comment:
What about adding this `filename` and `filetype` conversion to `uploadRemote` on `@uppy/tus` client instead? Then it’s consistent — tusd expects metadata this way, it doesn’t convert `name` to `filename` for you, and so will Companion. We have that conversion for direct uploads, and we’ll do the same in the same place for remote uploads.

https://github.com/transloadit/uppy/blob/master/packages/%40uppy/tus/src/index.js#L180-L184

Reply:
I guess we have to keep it in companion until 2.0 :(

Reply:
That’s right, forgot about that, good catch! But what about adding it on the client anyway, then we can add a 2.0 todo to remove on Companion and forget about it.

Reply:
> What about adding this filename and filetype conversion to uploadRemote on @uppy/tus client instead? Then it’s consistent 

@arturi but then we'd have to change it for XHR-Uploads too. And as you can see from the code, the way companion sets `filetype` for multipart is quite different and prone to change depending on the request library used. So I don't think it matters if we change it from the client. So long as companion knows what to expect, it can change it accordingly